compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -13599,36 +13599,9 @@ void arraycmpWithPadHelper::teardownCLC()
 
 void arraycmpWithPadHelper::generateCLCLitPoolPadding()
    {
-   TR::Compilation *comp = cg->comp();
-   bool constPadding = paddingNode->getOpCode().isLoadConst();
-   int32_t paddingByteLength = (source1Len > source2Len) ? (source1Len - source2Len) : (source2Len - source1Len);
-   uint8_t *paddingValue = (uint8_t *)comp->trMemory()->allocateHeapMemory(sizeof(uint8_t) * paddingByteLength);
-
-   if (constPadding)
-      {
-      if (isUTF16)
-         {
-         uint16_t paddingChar = paddingNode->getIntegerNodeValue<uint16_t>();
-         uint16_t *paddingArr = (uint16_t *)paddingValue;
-
-         for (int32_t i = 0; i < paddingByteLength/2; ++i)
-            paddingArr[i] = paddingChar;
-         }
-      else
-         {
-         uint8_t paddingChar = paddingNode->getIntegerNodeValue<uint8_t>();
-         memset(paddingValue, paddingChar, paddingByteLength);
-         }
-
-      constSpacePadding = true;
-
-      paddingLitPoolOffset = cg->findOrCreateLiteral(paddingValue, paddingByteLength);
-      newLitPoolReg = cg->isLiteralPoolOnDemandOn();
-      if (newLitPoolReg)
-         litPoolBaseReg = cg->allocateRegister();
-      else
-         litPoolBaseReg = cg->getLitPoolRealRegister();
-      }
+   // This path used to contain a call to an API which would have returned a garbage result. Rather than 100% of the
+   // time generating an invalid sequence here which is guaranteed to crash if executed, we fail the compilation.
+   cg->comp()->failCompilation<TR::CompilationException>("Existing code relied on an unimplemented API and is thus not safe. See eclipse/omr#5937.");
```

Comment:
This is safe because this class is only used in paths which generate `arraycmpWithPad` and this IL is slated for deprecation in #5914. This code will eventually be deleted.

Reply:
It seems @0xdaryl has removed this in #5941. Should I just remove this change and let you get rid of the whole thing. In this case my PR becomes dependent on yours.

Reply:
If you remove the change to break the dependency I am fine with that (assuming you are).

I have the downstream parts of this PR undergoing testing right now so the two PRs should be ready to review and merge soon.

Reply:
I've removed these changes from my PR. Thanks Daryl!